src/ux/table.ts

```diff
@@ -0,0 +1,321 @@
+
+      return {
+        extended,
+        get,
+        header,
+        key,
+        minWidth,
+      };
+    });
+
+    // assign options
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    const { columns: cols, extended, filter, printLine, sort, title } = options;
+    this.options = {
+      columns: cols,
+      extended,
+      filter,
+      'no-header': options['no-header'] ?? false,
+      'no-truncate': options['no-truncate'] ?? false,
+      printLine: printLine ?? ((s: string): void => ux.stdout(s)),
+      rowStart: ' ',
+      sort,
+      title,
+    };
+
+    // build table rows from input array data
+    let rows = data.map((d) =>
+      Object.fromEntries(
+        this.columns.map((col) => {
+          let val = col.get(d);
+          if (typeof val !== 'string') val = inspect(val, { breakLength: Number.POSITIVE_INFINITY });
+          return [col.key, val];
+        })
+      )
+    );
+
+    // filter rows
+    if (this.options.filter) {
+      // eslint-disable-next-line prefer-const
+      let [header, regex] = this.options.filter.split('=');
+      const isNot = header.startsWith('-');
+      if (isNot) header = header.slice(1);
+      const col = this.findColumnFromHeader(header);
+      if (!col || !regex) throw new Error('Filter flag has an invalid value');
+      rows = rows.filter((d) => {
+        const re = new RegExp(regex);
+        const val = d[col.key] as string;
+        const match = val.match(re);
+        return isNot ? !match : match;
+      });
+    }
+
+    // sort rows
+    if (this.options.sort) {
+      const sorters = this.options.sort.split(',');
+      const sortHeaders = sorters.map((k) => (k.startsWith('-') ? k.slice(1) : k));
+      const sortKeys = this.filterColumnsFromHeaders(sortHeaders).map(
+        (c) =>
+          (v: Record<string, unknown>): unknown =>
+            v[c.key]
+      );
+      const sortKeysOrder = sorters.map((k) => (k.startsWith('-') ? 'desc' : 'asc'));
+      rows = orderBy(rows, sortKeys, sortKeysOrder);
+    }
+
+    // and filter columns
+    if (this.options.columns) {
+      const filters = this.options.columns.split(',');
+      this.columns = this.filterColumnsFromHeaders(filters);
+    } else if (!this.options.extended) {
+      // show extended columns/properties
+      this.columns = this.columns.filter((c) => !c.extended);
+    }
+
+    this.data = rows;
+  }
+
+  public display(): void {
+    const { data, options } = this;
+    // column truncation
+    //
+    // find max width for each column
+    const columns = this.columns.map((c) => {
+      const maxWidth = Math.max(sw('.'.padEnd(c.minWidth - 1)), sw(c.header), getWidestColumnWith(data, c.key)) + 1;
+      return {
+        ...c,
+        maxWidth,
+        width: maxWidth,
+      };
+    });
+
+    // terminal width
+    const maxWidth = stdtermwidth - 2;
+    // truncation logic
+    const shouldShorten = (): void => {
```

Comment:
this reads like a boolean that makes a decision (should I shorten something) but is doing quite a bit.

Reply:
I'd probably call it "maybeShorten" or something clearer.

the flow of this is kinda odd (compute the columns as a const, then maybe go back and change them based on certain conditions)